packages/svelte/src/compiler/phases/utils.js

```diff
@@ -0,0 +1,39 @@
+/** @import { AST, Binding } from '#compiler' */
+
+/**
+ * Whether a variable can be referenced directly from template string.
+ * @param {Binding | undefined} binding
+ * @returns {boolean}
+ */
+function can_inline_variable(binding) {
+	return (
+		!!binding &&
+		// in a `<script module>` block
+		!binding.scope.parent &&
+		// to prevent the need for escaping
+		binding.initial?.type === 'Literal'
+	);
+}
+
+/**
+ * @param {(AST.Text | AST.ExpressionTag) | (AST.Text | AST.ExpressionTag)[]} node_or_nodes
+ * @param {import('./scope.js').Scope} scope
+ */
+export function is_inlinable_expression(node_or_nodes, scope) {
```

Comment:
this signature (`node_or_nodes`) feels like a code smell. can we change it to this?

```suggestion
 * @param {AST.Attribute} attribute
 * @param {Scope} scope
 */
export function is_inlinable_attribute(attribute, scope) {
```

(note that I got rid of the `import('...')` as well — we should use `/** @import ... */` instead)

Reply:
This was based on the initial code from @benmccann but let me see if it's doable

Reply:
Should be doable